### PR TITLE
Rework of judge

### PR DESCRIPTION
**Pull request by beardhatcode**
Made stuff nicer and error messages more clear. A bit of everything. 

Live at naos atm

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,60 +1,67 @@
...
 
-# Availible tests
+Regular
+[PLUnit](<https://www.swi-prolog.org/pldoc/doc_for?object=section(%27packages/plunit.html%27)>)
+can be used
 
 ## Form check
 
-The form check checks the form of submissions. This is absicly a linter
+The form check checks the form of submissions. This is basicly a linter.
```
Comment (charvp) [inline README.md]:
```suggestion
The form check checks the form of submissions. This is basically a linter.
```

File: README.md [hunk 2]
```diff
@@ -1,60 +1,67 @@
...
 
-## Todos
-
-- [x] Memory management
-- [ ] Using the partial format
-- [x] Add quickcheck (see the `feature/quickcheck` branch and PR)
-- [ ] make text below clearer
-
-## Format of the Quickcheck tests
+The name of the QuickCheck predicates should start with `prop_`
```
Comment (charvp) [inline README.md]: Obviously I'm not very familiar with Prolog, but isn't this the same as what you say on line 10?

File: README.md [hunk 3]
```diff
@@ -1,60 +1,67 @@
...
+    my_reverse(R, L).
+```
 
 ## Format of PLUnit testfiles
 
-The file will be splitted up in parts between the tags `:- begin_tests(...).` and `:- end_tests(...).`. All the lines specified between the parst above the current part wil be used.
+PLUnit testfiles should be named with a name ending in `.unit.pl`. The name
+before the extension will be used as tabname.
+
+The testfile will be splitted up in parts between the tags
```
Comment (charvp) [inline README.md]:
```suggestion
The testfile will be split up into the parts between the tags
```

File: README.md [hunk 4]
```diff
@@ -1,60 +1,67 @@
...
+`:- begin_tests(...).` and `:- end_tests(...).`. All the lines specified before
+the `begin_tests` and not in an other part can be used.
```
Comment (charvp) [inline README.md]:
```suggestion
the `begin_tests` and not in another part can be used.
```

Comment (charvp) [reply README.md]: Also not sure what you mean by, "can be used". I would assume it would say "will not be used".

File: README.md [hunk 5]
```diff
@@ -1,60 +1,67 @@
...
 ### Test specification
 
-It is not driven by a test file,
-but by the exercise configuration. Following keys can be inclused in the
-`evaluation` directive:
+It is not driven by a test file, but by the exercise configuration. Following
+keys can be inclused in the `evaluation` directive:
 
 - `cutallowed`: A boolean value indicationg if cuts are allowed (default false)
 - `predefined`: List of predefined predicates in `Name/Arrity` form, if this is
```
Comment (charvp) [inline README.md]:
```suggestion
- `predefined`: List of predefined predicates in `Name/Arity` form, if this is
```

File: README.md [hunk 6]
```diff
@@ -1,60 +1,67 @@
...
-   set, `check:check` will be executed (may be the empty list)
+  set, `check:check` will be executed (may be the empty list)
 
-The checkCheat test is always executed, it looks for obeccurences of
+The checkCheat test is always executed, it looks for ocurrences of
 `dodonaevaluate` and marks them as errorous.
```
Comment (charvp) [inline README.md]:
```suggestion
`dodonaevaluate` and marks them as erronous.
```

File: README.md [hunk 7]
```diff
@@ -81,41 +88,52 @@ The form is:
...
 ```
 
 The `test/6` predicate has 6 arguments
 
 1. The name, a string (double quotes), that is rendered as markdown
-2. The functor name of a correct implementation (mostly of the form `dodonaevaluate:....`)
+2. The functor name of a correct implementation (mostly of the form
+   `dodonaevaluate:....`)
 3. The functor name of the student implementation
 4. A Boolean value indicating is cuts are allowed (does not work properly yet)
```
Comment (charvp) [inline README.md]:
```suggestion
4. A Boolean value indicating whether cuts are allowed (does not work properly yet)
```

File: README.md [hunk 8]
```diff
@@ -81,41 +88,52 @@ The form is:
...
 %- consult private ["file1.pl","file2.pl"].
 ```
+
 ### Implementation
 
-The prolog checker code is contained in [simpletest/checker.pl](simpletest/checker.pl) it is loaded together
-with the student code as follows, with a copy of the workspace folder as workspace and `dotests` as 
-goals for `swipl`.
+The prolog checker code is contained in
+[simpletest/checker.pl](simpletest/checker.pl) it is loaded together with the
```
Comment (charvp) [inline README.md]:
```suggestion
[simpletest/checker.pl](simpletest/checker.pl). It is loaded together with the
```

File: README.md (hunk 2 above)
Comment (beardhatcode) [reply README.md]: yes

**Pull request merged**